### ISSUE TITLE
Transformer LM takes too long to infer on English!

### ISSUE THREAD

**Issue by neerajchhimwal** [pytorch/fairseq#2918]
#### What is your question?
Should transformer LM take 7 hours to infer on Libri Test-clean dataset?

Data: Libri Test-clean
LM checkpoint: transformer_lm.wiki103.adaptive (downloaded from repo)
Finetuning checkpoint: Wav2Vec 2.0 Large (LV-60)* 960h (downloaded from repo)

Time without Transformer lm: 244s for 2620 sentences (284022 tokens)
Time with Transformer lm: 7.4 hours for 2620 sentences (284022 tokens)

#### File/command used

python examples/speech_recognition/infer.py libri-test --task audio_pretraining --nbest 1 --path english_pretrained_large_checkpoint/wav2vec2_vox_960h_new.pt --gen-subset test --results-path libri/transcript --w2l-decoder fairseqlm --lm-model adaptive_lm_wiki103.v2/model.pt --lm-weight 2 --word-score -1 --sil-weight 0 --criterion ctc --labels ltr --max-tokens 640000 --post-process letter

#### What's your environment?

 - fairseq commit: 81677d75
 - OS (e.g., Linux): Linux
 - Python version: 3.7.9

**Comment by alexeib** [comment]
yeah sorry, it is not very efficient. i shard the eval jobs over multiple nodes to get it evaluated in a few hours (with a bigger transformer lm)
